packages/gguf/src/gguf.spec.ts

```diff
@@ -280,12 +282,47 @@ describe("gguf", () => {
 		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-Q2_K.gguf")).toEqual("Q2_K");
 		expect(parseGGUFQuantLabel("Codestral-22B-v0.1.gguf")).toEqual(undefined);
 		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-F32-Q2_K.gguf")).toEqual("Q2_K"); // gguf name with two quant labels [F32, Q2_K]
-		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-IQ3_XS.gguf")).toEqual(undefined); // TODO: investigate IQ3_XS
+		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-IQ3_XS.gguf")).toEqual("IQ3_XS");
 		expect(parseGGUFQuantLabel("Codestral-22B-v0.1-Q4_0_4_4.gguf")).toEqual("Q4_0"); // TODO: investigate Q4_0_4_4
 	});
 
 	it("calculate tensor data offset", async () => {
 		const { tensorDataOffset } = await gguf(URL_LLAMA);
 		expect(tensorDataOffset).toEqual(741056n);
 	});
+
+	// Quantization handler
+
+	it("should have GGUF_QUANT_ORDER in sync with GGMLQuantizationType enum", () => {
+		const enumValues = Object.values(GGMLQuantizationType).filter((value) => typeof value === "number") as number[];
+		const checkValues = new Set(GGUF_QUANT_ORDER);
+		for (const value of enumValues) {
+			expect(checkValues).toContain(value);
+		}
+	});
+
+	it("should find the nearest quant", () => {
+		const quant = GGMLFileQuantizationType.IQ2_M;
+		const availableQuants = [
+			GGMLFileQuantizationType.Q2_K,
+			GGMLFileQuantizationType.Q4_K_M,
+			GGMLFileQuantizationType.Q8_0,
+		];
+		const nearestQuant = findNearestQuantType(quant, availableQuants);
+		expect(nearestQuant).toEqual(GGMLFileQuantizationType.Q2_K);
+	});
+
+	it("should find the nearest quant (vision model)", () => {
+		const visionQuants = [GGMLFileQuantizationType.Q8_0, GGMLFileQuantizationType.F16, GGMLFileQuantizationType.BF16];
+		let nearestQuant;
+		// text = Q4_K_M
+		nearestQuant = findNearestQuantType(GGMLFileQuantizationType.Q4_K_M, visionQuants);
+		expect(nearestQuant).toEqual(GGMLFileQuantizationType.Q8_0);
+		// text = Q8_0
+		nearestQuant = findNearestQuantType(GGMLFileQuantizationType.Q8_0, visionQuants);
+		expect(nearestQuant).toEqual(GGMLFileQuantizationType.Q8_0);
+		// text = F16
+		nearestQuant = findNearestQuantType(GGMLFileQuantizationType.F16, visionQuants);
+		expect(nearestQuant).toEqual(GGMLFileQuantizationType.F16);
+	});
```

Comment:
Btw @bartowski1182 , this test case is inspired from a real world scenario where we have vision quantized to F16/BF16/Q8_0, and the text can be anything else.

Feel free to suggest other test cases if you can think of any!